akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -2351,8 +2351,16 @@ private[akka] final class StatefulMapConcat[In, Out](val f: () => In => Iterable
 
     override def onPush(): Unit =
       try {
-        currentIterator = plainFun(grab(in)).iterator
-        pushPull(shouldResumeContext = false)
+        val iterable = plainFun(grab(in))
+        if ((iterable.isEmpty): @nowarn("msg=deprecated")) {
```

Comment:
I don't think we should start using this as it was deprecated already for 2.13.0